src/watertap_contrib/seto/unit_models/zero_order/mg_crystallizer_zo.py

```diff
@@ -0,0 +1,595 @@
+        ConfigBlock(
+            implicit=True,
+            description="Arguments to use for constructing property packages",
+            doc="""A ConfigBlock with arguments to be passed to a property block(s)
+    and used when constructing these,
+    **default** - None.
+    **Valid values:** {
+    see property package for documentation.}""",
+        ),
+    )
+
+    def build(self):
+        super().build()
+
+        # TODO: add references
+
+        solvent_set = self.config.property_package.solvent_set
+        solute_set = self.config.property_package.solute_set
+
+        self.scaling_factor = Suffix(direction=Suffix.EXPORT)
+
+        units_meta = self.config.property_package.get_metadata().get_derived_units
+
+        # Add state blocks for inlet, outlet, and waste
+        # These include the state variables and any other properties on demand
+        # Add inlet block
+        tmp_dict = dict(**self.config.property_package_args)
+        tmp_dict["has_phase_equilibrium"] = False
+        tmp_dict["parameters"] = self.config.property_package
+        tmp_dict["defined_state"] = True
+        self.properties_in = self.config.property_package.state_block_class(
+            self.flowsheet().config.time, doc="Material properties of inlet", **tmp_dict
+        )
+
+        # Add outlet block
+        tmp_dict["defined_state"] = False
+        self.properties_out = self.config.property_package.state_block_class(
+            self.flowsheet().config.time,
+            doc="Material properties of liquid outlet",
+            **tmp_dict,
+        )
+        self.properties_out[0].pressure.fix(101325)
+        self.properties_out[0].flow_mass_phase_comp["Vap", "H2O"].fix(0)
+        self.properties_out[0].flow_mass_phase_comp["Sol", "NaCl"].fix(0)
+        # Initial guess
+        self.properties_out[0].flow_vol_phase["Liq"].value = 1e-5
+        self.properties_out[0].flow_vol_phase["Sol"].value = 1e-5
+
+        @self.Constraint(doc="Outlet temperature remains the same")
+        def eq_outlet_temperature(b):
+            return b.properties_out[0].temperature == b.properties_in[0].temperature
+
+        self.add_port(name="inlet", block=self.properties_in)
+        self.add_port(name="outlet", block=self.properties_out)
+
+        # Model parameters
+        self.hydromagnesite_solubility = Param(
+            initialize=0.76071,
+            units=pyunits.g / pyunits.L,
+            doc="Solubility of hydromagnesite [Mg5(CO3)4(OH)2 4H2O] at 25 deg C",
+        )
+
+        self.calcite_solubility = Param(
+            initialize=0.17631,
+            units=pyunits.g / pyunits.L,
+            doc="Solubility of calcite [CaCO3] at 25 deg C",
+        )
+
+        self.Mg_molar_weight = Param(
+            initialize=24.305, units=pyunits.g / pyunits.mol, doc="Molar weight of Mg"
+        )
+
+        self.Ca_molar_weight = Param(
+            initialize=40.087, units=pyunits.g / pyunits.mol, doc="Molar weight of Mg"
+        )
```

Comment:
These are OK as Params since we would never fit these with parameter estimation. However, if using MCAS, you could provide these up front when creating an instance of MCAS.